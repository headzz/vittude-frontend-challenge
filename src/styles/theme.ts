export const theme = {
  colors: {
    primary: '#FF6952',
    secondary: '#F26B5B',
    error: '#F63D15',
    dark: {
      100: '#777C88',
      200: '#2E3035',
      300: '#000000',
    },
    light: {
      100: '#FFFFFF',
      200: '#E7EAF4',
    },
  },
  font: {
    size: {
      normal: '1rem',
      small: '0.875rem',
      large: '2rem',
      extraLarge: '3rem',
    },
    type: { header: 'Inter', normal: 'Fira Sans' },
  },
  spacing: {
    extraSmall: '10px',
    small: '12px',
    normal: '16px',
    large: '24px',
    extraLarge: '48px',
  },
}
