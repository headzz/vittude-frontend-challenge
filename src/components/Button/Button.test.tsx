import { fireEvent, render, screen } from '@testing-library/react'
import { Button } from '.'
import { ThemeProvider } from 'styled-components'
import { theme } from '@styles/theme'

describe('Button', () => {
  const mockClick = jest.fn()
  it('renders primary button', () => {
    render(
      <ThemeProvider theme={theme}>
        <Button handleClick={mockClick} primary>
          Test
        </Button>
      </ThemeProvider>,
    )

    const button = screen.getByRole('button')

    expect(button).toBeInTheDocument()
    expect(button).toHaveTextContent('Test')
    expect(button).toMatchSnapshot()
  })

  it('should trigger click', () => {
    render(
      <ThemeProvider theme={theme}>
        <Button handleClick={mockClick} primary>
          Test
        </Button>
      </ThemeProvider>,
    )

    const button = screen.getByRole('button')
    fireEvent.click(button)

    expect(mockClick).toBeCalled()
  })

  it('should render disabled button', () => {
    render(
      <ThemeProvider theme={theme}>
        <Button handleClick={mockClick} primary disabled>
          Test
        </Button>
      </ThemeProvider>,
    )

    const button = screen.getByRole('button')

    expect(button).toBeInTheDocument()
    expect(button).toHaveTextContent('Test')
    expect(button).toMatchSnapshot()
  })

  it('should render outline button', () => {
    render(
      <ThemeProvider theme={theme}>
        <Button handleClick={mockClick} primary={false} disabled>
          Test
        </Button>
      </ThemeProvider>,
    )

    const button = screen.getByRole('button')

    expect(button).toBeInTheDocument()
    expect(button).toHaveTextContent('Test')
    expect(button).toMatchSnapshot()
  })
})
