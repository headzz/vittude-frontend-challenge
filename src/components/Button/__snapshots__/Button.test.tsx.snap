// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders primary button 1`] = `
<button
  class="sc-bcPKhP xeGTD"
>
  Test
</button>
`;

exports[`Button should render disabled button 1`] = `
<button
  class="sc-bcPKhP xeGTD"
  disabled=""
>
  Test
</button>
`;

exports[`Button should render outline button 1`] = `
<button
  class="sc-bcPKhP bEYXBt"
  disabled=""
>
  Test
</button>
`;
